src/common/reducers/repositories.js

```diff
@@ -2,10 +2,11 @@ import * as ActionTypes from '../actions/repositories';
 import { parseGitHubRepoUrl } from '../helpers/github-url';
 
 export function repositories(state = {
-  isFetching: false,
+  // has no server side content, so initial state is always 'fetching'
+  isFetching: true,
```

Comment:
I don't think it matters when it is fetching, we could dispatch this action after the request is initiated if we wanted, and it would still fulfil its purpose: communicating to the user that the application is waiting for data it doesn't have. Other applications use props named "waiting" or "pending".

I think it's still worth having this discussion, because this is a good example of state that will be re-used wherever we have a component that waits for data. Sooner or later we'll need to have the conversation again. Here's my recommendation:

- **pending**: Component is waiting for new data when true
- **initialised**: Component needs new data, but cannot request it because it is rendered on server or waiting for user interaction when false.
- **last-requested**: A timestamp of when the last request started. Used for debouncing API requests.